fuel-core-interfaces/src/model/messages.rs

```diff
@@ -0,0 +1,65 @@
+use super::BlockHeight;
+use core::ops::Deref;
+use fuel_crypto::Hasher;
+use fuel_types::{Address, Bytes32, Word};
+
+/// Message send from Da layer to fuel by bridge
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct DaMessage {
+    pub sender: Address,
+    pub receipient: Address,
+    pub owner: Address,
+    pub nonce: Word,
+    pub amount: Word,
+    pub data: Vec<u8>,
+    pub fuel_block_spend: Option<BlockHeight>,
+}
+
+impl DaMessage {
+    fn id(&self) -> Bytes32 {
+        let mut hasher = Hasher::default();
+        hasher.input(self.sender);
+        hasher.input(self.receipient);
+        hasher.input(self.owner);
+        hasher.input(self.nonce.to_be_bytes());
+        hasher.input(self.amount.to_be_bytes());
+        hasher.input(&self.data);
+        hasher.digest()
+    }
+
+    pub fn lock(self) -> DaMessageLocked {
+        let id = self.id();
+        DaMessageLocked { message: self, id }
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct DaMessageLocked {
```

Comment:
Hm i used here `Locked` naming, while Transaction has `Checked`.
Idea is that we `lock` the data that is not possible to be changed in a locked state (It has `AsRef` and `unlock` function). Are we okay to have two namings or should i rename it to `Checked` for consistency?

Reply:
If they're the same thing, then we should make it a trait and name them under a similar convention both implementing that trait. If they don't behave similar enough to share a trait, then I don't have a problem with them just being named in a different style

Reply:
Lock may be confusing with concurrency related issues. Since we've already discussed using "Checked" types before, and have already used that convention in fuel-tx and fuel-vm for some time, I'd prefer to keep things consistent with Checked. Although if you think Checked isn't a great name, then we should have a team vote and then change it in all the other places to be consistent.